.github/prereleases/0-packages.mjs

```diff
@@ -0,0 +1,108 @@
+import assert from "node:assert";
+import { execSync } from "node:child_process";
+import fs from "node:fs";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+export const projectRoot = path.resolve(__dirname, "../..");
+
+const githubRunId = parseInt(process.env.GITHUB_RUN_ID);
+assert(
+	!Number.isNaN(githubRunId),
+	"Expected GITHUB_RUN_ID variable to be a number"
+);
+const githubEventPath = process.env.GITHUB_EVENT_PATH;
+assert(githubEventPath, "Expected GITHUB_EVENT_PATH variable to be defined");
+const githubEventContents = fs.readFileSync(githubEventPath, "utf8");
+const githubEvent = JSON.parse(githubEventContents);
+assert(
+	typeof githubEvent === "object" &&
+		githubEvent !== null &&
+		"pull_request" in githubEvent &&
+		typeof githubEvent.pull_request === "object" &&
+		githubEvent.pull_request !== null &&
+		"number" in githubEvent.pull_request &&
+		typeof githubEvent.pull_request.number === "number",
+	`Expected valid pull_request event, got ${githubEventContents}`
+);
+// noinspection JSUnresolvedReference,JSObjectNullOrUndefined
+const githubPullRequestNumber = githubEvent.pull_request.number;
+
+/**
+ * @typedef {object} ~PackageJsonWorkersSdk
+ * @property {boolean} [prerelease]
+ */
+
+/**
+ * @typedef {object} ~PackageJson
+ * @property {string} name
+ * @property {string} version
+ * @property {Record<string, string>} [dependencies]
+ * @property {Record<string, string>} [devDependencies]
+ * @property {Record<string, string>} [peerDependencies]
+ * @property {Record<string, string>} [optionalDependencies]
+ * @property {~PackageJsonWorkersSdk} [workers-sdk]
+ */
+
+/**
+ * @typedef {object} ~Package
+ * @property {string} path
+ * @property {~PackageJson} json
+ */
+
+/** @returns {string[]} */
+function getPackagePaths() {
+	const stdout = execSync(
+		'pnpm list --filter="./packages/*" --recursive --depth=-1 --parseable',
```

Comment:
When I run this command locally I get a lot of false positives. E.g.

```
/Users/pbacondarwin/dev/cloudflare/workers-sdk/packages/wrangler
/Users/pbacondarwin/dev/cloudflare/workers-sdk/node_modules/.pnpm/@babel+core@7.22.5_supports-color@9.2.2/node_modules/@babel/core
/Users/pbacondarwin/dev/cloudflare/workers-sdk/node_modules/.pnpm/@babel+plugin-transform-modules-commonjs@7.22.5_@babel+core@7.22.5_supports-color@9.2.2/node_modules/@babel/plugin-transform-modules-commonjs
/Users/pbacondarwin/dev/cloudflare/workers-sdk/node_modules/.pnpm/@babel+runtime@7.22.5/node_modules/@babel/runtime
/Users/pbacondarwin/dev/cloudflare/workers-sdk/node_modules/.pnpm/@cloudflare+ai@1.0.41/node_modules/@cloudflare/ai
/Users/pbacondarwin/dev/cloudflare/workers-sdk/packages/cli
/Users/pbacondarwin/dev/cloudflare/workers-sdk/packages/eslint-config-worker
/Users/pbacondarwin/dev/cloudflare/workers-sdk/node_modules/.pnpm/@cloudflare+intl-types@1.2.0_react@17.0.2/node_modules/@cloudflare/intl-types
/Users/pbacondarwin/dev/cloudflare/workers-sdk/node_modules/.pnpm/@cloudflare+kv-asset-handler@0.2.0/node_modules/@cloudflare/kv-asset-handler
/Users/pbacondarwin/dev/cloudflare/workers-sdk/packages/pages-shared
/Users/pbacondarwin/dev/cloudflare/workers-sdk/node_modules/.pnpm/@cloudflare+types@6.18.4_react@17.0.2/node_modules/@cloudflare/types
/Users/pbacondarwin/dev/cloudflare/workers-sdk/node_modules/.pnpm/@cloudflare+util-en-garde@8.0.10/node_modules/@cloudflare/util-en-garde
/Users/pbacondarwin/dev/cloudflare/workers-sdk/packages/workers-tsconfig
```

Is that expected and or desired?

Reply:
I appreciate that the will all be filtered out by the `getPackagesForPrerelease()` function but still...

Reply:
Oh! I just realised you are doing depth is -1 (I read it as 1).